kernel/dw1000.c

```diff
@@ -1435,6 +1443,91 @@ static void dw1000_timestamp(struct dw1000 *dw,
 #endif
 }
 
+/**
+ * dw1000_rx_link_qual() - Check RX quality and calculate KPIs
+ *
+ * @dw:			DW1000 device
+ */
```

Comment:
I think the algorithm here is based on the DW1000 datasheet.  It's probably worth mentioning this (with source document references) in a comment, in case someone in future wonders why we're doing this calculation.